File: implementations/java/TSPData.java
public class TSPData {
    private int n_;
    private int[] d_;
    // auxiliary graphs used for espprc and maxflow
    private double[] aux_;
    private double[] aux2_;
    
    // here be getters
    public int d(int i, int j) {
	return d_[i*n_+j];
    }
    public int n() { return n_; }
    public int[] d() { return d_; }
    public double[] aux() { return aux_; }
    public double[] aux2() { return aux2_; }
    
    public TSPData(int n, int[]d) {
	n_ = n;
	d_ = d;
        aux_ = new double[n*n];
        aux2_ = new double[n*n];
    }
}

File: implementations/numpy/tspdata.py
import sys
import copy
import random
import math

import numpy

mapsize = 100

# encapsulate TSP data: instance + starting solutions to optimise for
# benchmarking purpose
class TSPData:
    def __init__(self, n, d):
        self.n, self.d = n, numpy.array(d)
        # auxiliary graph used e.g. for storing reduced costs (espprc)
        # or flow values (maxflow)
        # Typically it is used many times but only needs to be allocated once
        # and represents some kind of input data, therefore this is a good
        # place to have it.
        # dummy values to begin with...
        self.aux = numpy.array([ [ float(x) for x in row ] for row in d ])
        self.aux2 = numpy.array([ [ float(x) for x in row ] for row in d ])
